client/src/views/workspaces/WorkspaceSharingModal.vue

```diff
@@ -191,11 +267,73 @@ function isSelectDisabled(role: WorkspaceRole | null): boolean {
   return false;
 }
 
+async function onBatchSharingActivate(): Promise<void> {
+  if (showCheckboxes.value === true) {
+    for (const user of userRoles.value) {
+      user.isSelected = false;
+    }
+    membersCheckbox.value = false;
+  }
+  showCheckboxes.value = !showCheckboxes.value;
+}
+
+function getLowestSelectedProfile(): UserProfile {
+  let lowestProfile = UserProfile.Admin;
+  for (const user of selectedUsers.value) {
+    if (user.role[0].profile === UserProfile.Outsider) {
+      return UserProfile.Outsider;
+    }
+    if (user.role[0].profile === UserProfile.Standard) {
+      lowestProfile = UserProfile.Standard;
+    }
+  }
+  return lowestProfile;
+}
+
+function getHighestSelectedRole(): WorkspaceRole {
+  let highestRole = WorkspaceRole.Reader;
+  for (const user of selectedUsers.value) {
+    if (!user.role[1]) {
+      continue;
+    }
+    if (user.role[1] in [WorkspaceRole.Manager, WorkspaceRole.Owner]) return user.role[1];
+    if (user.role[1] === WorkspaceRole.Contributor) {
+      highestRole = WorkspaceRole.Contributor;
+    }
+  }
+  return highestRole;
+}
+
+const options = computed((): MsOptions => {
+  return new MsOptions(
+    [WorkspaceRole.Owner, WorkspaceRole.Manager, WorkspaceRole.Contributor, WorkspaceRole.Reader, null].map(
+      (role: WorkspaceRole | null) => {
+        return {
+          key: role,
+          label: getWorkspaceRoleTranslationKey(role).label,
+          description: getWorkspaceRoleTranslationKey(role).description,
+          disabled: !canChangeRole(ownProfile, getLowestSelectedProfile(), props.ownRole, getHighestSelectedRole(), role).authorized,
+          disabledReason: canChangeRole(ownProfile, getLowestSelectedProfile(), props.ownRole, getHighestSelectedRole(), role).reason,
+        };
+      },
+    ),
+  );
+});
+
+function selectAllMembers(): void {
+  for (const user of filteredSharedUserRoles.value) {
+    user.isSelected = !membersCheckbox.value;
+  }
+}
+
 async function refreshSharingInfo(): Promise<void> {
   const result = await getWorkspaceSharing(props.workspaceId, true);
 
   if (result.ok) {
-    userRoles.value = result.value;
+    userRoles.value = [];
```

Comment:
You can use `.map` to convert an array of object to an array of other objects (or to add a property)